src/handle_connect.c

```diff
@@ -457,7 +457,11 @@ int handle__connect(struct mosquitto_db *db, struct mosquitto *context)
 						rc = 1;
 						goto handle_connect_error;
 					}
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
the thing is, this only handles the _version_ there's also the OPENSSL_COMPAT_ defines, for prople compiling with deprecated functions disabled.